website/views.py

```diff
@@ -23,4 +29,129 @@ def oh(request):
+    date = datetime.day
+
+    current_week = get_dates_of_week(now)
+    print(current_week)
+
+    context = {'interview_slot_list': interview_slot_list, 'day': now.strftime("%A"), 'date': date, 
+    "time_dict": time_dict, "time_slot_dict": time_slot_dict, "start_times": start_times, 
+    "range": range(len(start_times)), "week": current_week}
+
+    return render(request, 'website/interview.html', context)
+
+def get_dates_of_week(now):
+    this_week = ['date' for i in range(7)]
+    current_day = now.weekday()
+    if current_day == 6:
+        this_week[0] = now.strftime('%b %d, %Y')
+        for i in range(1, 7):
+            add_date = now + timedelta(days=i)
+            this_week[i] = add_date.strftime('%b %d, %Y')
+    else:
+        num_things_before = current_day + 1
+        num_things_after = 5 - current_day
+        sunday = now - timedelta(days=current_day + 1)
+        this_week[0] = sunday.strftime('%b %d, %Y')
+
+        for i in range(0, current_day):
+            diff = current_day - i
+            add_date = now - timedelta(days=diff)
+            this_week[i + 1] = add_date.strftime('%b %d, %Y')
+        for j in range(current_day + 1, 7):
+            diff = j - current_day - 1           
+            add_date = now + timedelta(days=diff)
+            this_week[j] = add_date.strftime('%b %d, %Y')
+    return this_week
+
+def book_interview(request, slot_id):
+    all_slots = InterviewSlot.objects.all()
+    for slot in all_slots:
+        if slot.slot_id == slot_id:
+            context = {'time_slot': slot}
+            break
+    return render_to_response('website/book_interview.html', RequestContext(request, context))
+
+def confirm_interview(request):
+    if request.method == 'POST':
+
+        name = request.POST.get('name')
+        email = request.POST.get('email')
+
+        all_slots = InterviewSlot.objects.all()
+        num_student_has = len(all_slots.filter(student=name))
+        if num_student_has < 2 and validate_name(name) and validate_berkeley_email(email):
+            booked_slot = None
+            for slot in all_slots:
+                if slot.get_date() == request.POST['date'] and slot.hour == int(request.POST['day_hour'][1:]):
+                    booked_slot = slot
+                    break
+            #if booked_slot == None:
+            #   return oh(request)
+            booked_slot.student = request.POST['name']
+            booked_slot.student_email = request.POST['email']
+            booked_slot.availability = False
+            booked_slot.save()
+            send_confirmation_email(booked_slot)
+            return interview(request)
+        else:
+            return interview(request)       
+    else:
+        return interview(request)
+
+def validate_name(name):
+    return name is not None
+
+def validate_berkeley_email(email):
+    try:
+        validate_email(email)
+    except ValidationError as e:
+        return False
+    else:
+        if "@berkeley.edu" not in email:
```

Comment:
if you were worried about the @berkeley.edu@some_other_domain.com, you can use regex to parse this instead.
That way you can make sure the email is "@berkeley.edu" followed by the end of the string. But not necessary